### PR TITLE
Rollback service startup on error

### PR DESCRIPTION
**Pull request by Cerv1**
This PR solves the issue number #1260

### CODE REVIEW

File: src/client-agent/main.c [hunk 1]
```diff
@@ -164,6 +164,9 @@ int main(int argc, char **argv)
         merror_exit(USER_ERROR, user, group);
     }
 
+    /* Check client keys */
+    OS_ReadKeys(&keys, 1, 0, 0);
```
Comment (chemamartinez) [inline src/client-agent/main.c]: If we call this function here, we should avoid to call it again in AgentdStart() some lines below.

It also has to have the same behavior on Windows.

File: src/shared/file_op.c [hunk 2]
```diff
@@ -474,7 +474,7 @@ int CreatePID(const char *name, int pid)
     }
 
     fprintf(fp, "%d\n", pid);
-
+    fflush(fp);
```
Comment (chemamartinez) [inline src/shared/file_op.c]: Why is this necessary?

Comment (Cerv1) [reply src/shared/file_op.c]: It wasn't strictly necessary. We wanted to increase the speed of creation of the file containing the PID but it didn't make a difference. I'll remove this line.

File: src/client-agent/main.c (hunk 1 above)
Comment (Cerv1) [reply src/client-agent/main.c]: Good catch, I didn't see that! Thanks!

**Comment by chemamartinez** [comment]
Done at #1324 for the branch 3.6.
